packages/track-event-mixin/track-event-gesture.ts

```diff
@@ -0,0 +1,308 @@
+function isMouseEvent(name: string): boolean {
+  return ['mousedown', 'mousemove', 'mouseup'].indexOf(name) > -1;
+}
+
+function hasLeftMouseButton(ev: MouseEvent): boolean {
+  const { type } = ev;
+  let result;
+  if (type === 'mousemove') {
+    // ev.button is not reliable for mousemove (0 is overloaded as both left button and no buttons)
+    // instead we use ev.buttons (bitmask of buttons): 0 for no buttons, 1 for left button
+    // NOTE: allow undefined for testing events
+    const buttons = ev.buttons === undefined ? 1 : ev.buttons;
+    // buttons is a bitmask, check that the left button bit is set (1)
+    result = Boolean(buttons & 1); // eslint-disable-line no-bitwise
+  } else {
+    // NOTE: allow undefined for testing events
+    const button = ev.button === undefined ? 0 : ev.button;
+    // ev.button is 0 in mousedown/mouseup/click for left button activation
+    result = button === 0;
+  }
+  return result;
+}
+
+function trackHasMovedEnough(info: TrackGestureInfo, x: number, y: number): boolean {
+  if (info.started) {
+    return true;
+  }
+  const dx = Math.abs(info.x - x);
+  const dy = Math.abs(info.y - y);
+  return dx >= TRACK_DISTANCE || dy >= TRACK_DISTANCE;
+}
+
+function trackFire(info: TrackGestureInfo, target: EventTarget, touch: Touch | MouseEvent) {
+  const { x, y, moves, state } = info;
+  const secondLast = moves[moves.length - 2];
+  const lastMove = moves[moves.length - 1];
+  const dx = lastMove.x - x;
+  const dy = lastMove.y - y;
+  let ddx = 0;
+  let ddy = 0;
+  if (secondLast) {
+    ddx = lastMove.x - secondLast.x;
+    ddy = lastMove.y - secondLast.y;
+  }
+  target.dispatchEvent(
+    new CustomEvent('track', {
+      bubbles: true,
+      cancelable: true,
+      composed: true,
+      detail: {
+        state,
+        x: touch.clientX,
+        y: touch.clientY,
+        dx,
+        dy,
+        ddx,
+        ddy,
+        sourceEvent: touch
+      }
+    })
+  );
+}
+
+function trackDocument(
+  info: TrackGestureInfo,
+  moveFn: (event: MouseEvent) => void,
+  upFn: (event: MouseEvent) => void
+) {
+  info.moveFn = moveFn;
+  info.upFn = upFn;
+  document.addEventListener('mousemove', moveFn);
+  document.addEventListener('mouseup', upFn);
+}
+
+function unTrackDocument(info: TrackGestureInfo) {
+  document.removeEventListener('mousemove', info.moveFn as EventListener);
+  document.removeEventListener('mouseup', info.upFn as EventListener);
+  info.moveFn = undefined;
+  info.upFn = undefined;
+}
+
+const TrackGesture: TrackGesture = {
+  name: 'track',
+  touchAction: 'none',
+  deps: ['mousedown', 'touchstart', 'touchmove', 'touchend'],
+  info: {
+    x: 0,
+    y: 0,
+    state: 'start',
+    started: false,
+    moves: [],
+    addMove(move) {
+      if (this.moves.length > TRACK_LENGTH) {
+        this.moves.shift();
+      }
+      this.moves.push(move);
+    },
+    moveFn: undefined,
+    upFn: undefined
+  },
+
+  /**
+   * @this {GestureRecognizer}
+   * @return {void}
+   */
```

Comment:
Should this comment be removed? This seems to be the only reference to `GestureRecognizer` and these annotations seem unnecessary anyway or is there a purpose to them?

Reply:
Thanks, removed.